rtgui/colortoning.cc

```diff
@@ -336,6 +339,116 @@ ColorToning::ColorToning () : FoldableToolPanel(this, "colortoning", M("TP_COLOR
     pack_start(*labgridBox, Gtk::PACK_EXPAND_WIDGET, 4);
     //------------------------------------------------------------------------
 
+    //------------------------------------------------------------------------
+    // LAB regions
+
+    const auto add_button =
+        [&](Gtk::Button *btn, Gtk::Box *box) -> void
+        {
+            setExpandAlignProperties(btn, false, false, Gtk::ALIGN_CENTER, Gtk::ALIGN_START);
+            btn->set_relief(Gtk::RELIEF_NONE);
+            btn->get_style_context()->add_class(GTK_STYLE_CLASS_FLAT);
+            btn->set_can_focus(false);
+            btn->set_size_request(-1, 20);
+            box->pack_start(*btn, false, false);
+        };
+    
+    EvLabRegionList = m->newEvent(LUMINANCECURVE, "HISTORY_MSG_COLORTONING_LABREGION_LIST");
+    EvLabRegionAB = m->newEvent(LUMINANCECURVE, "HISTORY_MSG_COLORTONING_LABREGION_AB");
+    EvLabRegionSaturation = m->newEvent(LUMINANCECURVE, "HISTORY_MSG_COLORTONING_LABREGION_SATURATION");
+    EvLabRegionLightness = m->newEvent(LUMINANCECURVE, "HISTORY_MSG_COLORTONING_LABREGION_LIGHTNESS");
+    EvLabRegionHueMask = m->newEvent(LUMINANCECURVE, "HISTORY_MSG_COLORTONING_LABREGION_HUEMASK");
+    EvLabRegionChromaticityMask = m->newEvent(LUMINANCECURVE, "HISTORY_MSG_COLORTONING_LABREGION_CHROMATICITYMASK");
+    EvLabRegionLightnessMask = m->newEvent(LUMINANCECURVE, "HISTORY_MSG_COLORTONING_LABREGION_LIGHTNESSMASK");
+    EvLabRegionShowMask = m->newEvent(LUMINANCECURVE, "HISTORY_MSG_COLORTONING_LABREGION_SHOWMASK");
+    labRegionBox = Gtk::manage(new Gtk::VBox());
+
+    labRegionList = Gtk::manage(new Gtk::ListViewText(3));
+    labRegionList->set_size_request(-1, 100);
+    labRegionList->set_column_title(0, "#");
+    labRegionList->set_column_title(1, M("TP_COLORTONING_LABREGION_LIST_TITLE"));
+    labRegionList->set_column_title(2, M("TP_COLORTONING_LABREGION_MASK"));
+    labRegionList->set_activate_on_single_click(true);
+    labRegionSelectionConn = labRegionList->get_selection()->signal_changed().connect(sigc::mem_fun(this, &ColorToning::onLabRegionSelectionChanged));
+    Gtk::HBox *hb = Gtk::manage(new Gtk::HBox());
```

Comment:
Please be so kind to constify those pointers, as we had [problems](https://github.com/Beep6581/RawTherapee/commit/653f252b9c0bac8366e18e7cb2ebf345d0817d2c) with large sequences of assignments in the past.